components/ILIAS/Authentication/classes/Logout/class.ilAuthLogoutBehaviourGUI.php

```diff
@@ -0,0 +1,213 @@
+                    fn(callable $txt, $value) => $txt('logout_behaviour_invalid_ref_id', $value)
+                )
+            )
+            ->withAdditionalTransformation(
+                $this->refinery->custom()->constraint(
+                    fn($value) => $this->access->checkAccessOfUser(ANONYMOUS_USER_ID, 'read', '', $value),
+                    fn(callable $txt, $value) => $txt('logout_behaviour_ref_id_no_access', $value)
+                )
+            )
+            ->withValue($this->settings->get('logout_behaviour_ref_id', ''));
+        if (isset($errors['ref_id'])) {
+            $ref_id = $ref_id->withError($errors['ref_id']);
+        }
+
+        $internal_group = $this->ui_factory->input()->field()
+            ->group(
+                ['ref_id' => $ref_id],
+                $this->lng->txt('destination_internal_ressource')
+            );
+
+        $url = $this->settings->get('logout_behaviour_url', '');
+        $html = $this->ui_factory->input()->field()
+            ->url($this->lng->txt('destination_external_ressource_url'))
+            ->withAdditionalTransformation(
+                $this->refinery->custom()->constraint(
+                    fn($value) => filter_var($value, FILTER_VALIDATE_URL) !== false,
+                    fn(callable $txt, $value) => $txt('logout_behaviour_invalid_url', $value)
+                )
+            )
+            ->withValue($url);
+        if (isset($errors['url'])) {
+            $html = $html->withError($errors['url']);
+        }
+
+        $external_group = $this->ui_factory->input()->field()
+            ->group(
+                ['url' => $html],
+                $this->lng->txt('destination_external_ressource')
+            );
+
+        $logout_behaviour_switchable_group = $this->ui_factory->input()->field()
+            ->switchableGroup(
+                [
+                    'logout_screen' => $logout_group,
+                    'login_screen' => $login_group,
+                    'internal_ressource' => $internal_group,
+                    'external_ressource' => $external_group
+                ],
+                $this->lng->txt('destination_after_logout')
+            )
+            ->withValue($this->settings->get('logout_behaviour', 'logout_screen'));
+
+        $section = $this->ui_factory->input()->field()
+            ->section(
+                ['logout_behaviour_settings' => $logout_behaviour_switchable_group],
+                $this->lng->txt('logout_behaviour_settings')
+            );
+
+        $form = $this->ui_factory->input()->container()->form()
+            ->standard(
+                $this->ctrl->getFormAction($this, 'saveForm'),
+                ['logout_behaviour' => $section]
+            );
+        if ($request) {
+            $form = $form->withRequest($request);
+        }
+
+        return $form;
+    }
+
+    public function showForm(): void
+    {
+        $this->tpl->setContent($this->ui_renderer->render($this->getForm()));
```

Comment:
If we show the form with an existing configuratrion and the entered `$red_id` is not valid anymore, we should provide a `MessageBox > Failure` here.